tests/sympc/tensor/mpc_tensor_test.py

```diff
@@ -0,0 +1,208 @@
+# stdlib
+import operator
+
+# third party
+import pytest
+import torch
+
+from sympc.session import Session
+from sympc.tensor import MPCTensor
+
+
+def test_mpc_tensor_exception(clients) -> None:
+    alice_client, bob_client = clients
+    # TODO: for more than 2 parties
+    session = Session(parties=[alice_client, bob_client])
+
+    with pytest.raises(ValueError):
+        MPCTensor(secret=42, session=session)
+
+
+def test_reconstruct(clients) -> None:
+    alice_client, bob_client = clients
+    # TODO: for more than 2 parties
+    session = Session(parties=[alice_client, bob_client])
+    Session.setup_mpc(session)
+
+    x_secret = torch.Tensor([1, -2, 3.0907, -4.870])
+    x = MPCTensor(secret=x_secret, session=session)
+    x = x.reconstruct()
+
+    assert torch.allclose(x_secret, x)
+
+
+def test_sessions_without_setup_mpc(clients):
+    alice_client, bob_client = clients
+    session = Session(parties=[alice_client, bob_client])
+
+    with pytest.raises(ValueError):
+        x = MPCTensor(secret=torch.Tensor([1, -2]), session=session)
+
+
+def test_sessions_with_different_sessions(clients):
+    alice_client, bob_client = clients
+    session_one = Session(parties=[alice_client, bob_client])
+    session_two = Session(parties=[alice_client, bob_client])
+    Session.setup_mpc(session_one)
+    Session.setup_mpc(session_two)
+
+    x = MPCTensor(secret=torch.Tensor([1, -2]), session=session_one)
+    y = MPCTensor(secret=torch.Tensor([1, -2]), session=session_two)
+
+    with pytest.raises(ValueError):
+        z = x + y
+
+
+def test_remote_mpc_no_shape(clients):
+    alice_client, bob_client = clients
+    session = Session(parties=[alice_client, bob_client])
+    Session.setup_mpc(session)
+
+    x_remote = alice_client.torch.Tensor([1, -2, 0.3])
+
+    with pytest.raises(ValueError):
+        x = MPCTensor(secret=x_remote, session=session)
+
+
+def test_remote_mpc_with_shape(clients):
+    alice_client, bob_client = clients
+    session = Session(parties=[alice_client, bob_client])
+    Session.setup_mpc(session)
+
+    x_remote = alice_client.torch.Tensor([1, -2, 0.3])
+    x = MPCTensor(secret=x_remote, shape=(1, 3), session=session)
+    result = x.reconstruct()
+
+    assert x_remote == result
+
+
+def test_remote_not_tensor(clients):
+    alice_client, bob_client = clients
+    session = Session(parties=[alice_client, bob_client])
+    Session.setup_mpc(session)
+
+    x_remote_int = bob_client.python.Int(5)
+    x = MPCTensor(secret=x_remote_int, shape=(1,), session=session)
+    result = x.reconstruct()
```

Comment:
Q: When we reconstruct here, ```result``` is an integer?

Reply:
no